gist.el

```diff
@@ -186,7 +194,7 @@ With a prefix argument, makes a private paste."
 
 (defun gist-created-callback (gist)
   (let ((location (oref gist :html-url)))
-    (gist-list-reload t)
+    (gist-list-reload "" t)
```

Comment:
The problem is that we have to differentiate between many cases:

* gist-list-user was called...
  * by the user with M-x gist-list-user
    * ...and the user provided a username
    * ...and the user did not provide a username (read-from-minibuffer returns "" in this case)
  * by gist-list-reload
    * ...because the user pressed "g" in a gist list buffer, so we want to reload that buffer
    * ...from gist-created-callback, so we want to reload the user's own gist list
  * by gist-list, so we should show the user's own gist list (for compatibility I figured we should not change the behavior or signature of `gist-list')
  * maybe one or two more I am forgetting :)

I also wanted to make sure it was still easy to use gist.el programmatically (other functions could be improved wrt this but I limited this patch to only the new features so that the diff would not have irrelevant changes).